Add ability to get and set Task name (#4777)
cocotb/cocotb: docs/source/newsfragments/4777.feature.1.rst

> Added the ``name`` keyword argument to :func:`cocotb.start_soon`, :func:`cocotb.start`, and :func:`cocotb.create_task` to set a :class:`.Task`\ s name when constructing one.

Author: ```suggestion
Added the ``name`` keyword argument to :func:`cocotb.start_soon`, :func:`cocotb.start`, and :func:`cocotb.create_task` to set a :class:`.Task`\ s name during creation.
```
